backend/app/user/supabase_user_service.py

```diff
@@ -0,0 +1,158 @@
+from typing import cast, override
+from uuid import UUID
+
+from fastapi import HTTPException, status
+
+from app.auth.supabase import SupabaseSignInAnonymouslyResult, SupabaseUserResult
+from app.auth.supabase import supabase_auth_service as auth_service
+from app.constants import ARKG_ANTI_RECOMMENDER_USER_STATE_TABLE_NAME
+from app.database.exceptions import DatabaseException
+from app.database.supabase import supabase_database_service
+from app.models import TableQuery, Token
+from app.models.types import RecordKey
+from app.user import User, UserService
+
+
+class SupabaseUserService(UserService):
+    """
+    A concrete implementation of `UserService`.
+
+    A `SupabaseUserService` uses a Supabase database service to manage the state of a `User`.
```

Comment:
OK, I see why DatabaseService exists now.

I think you could just dispense with DatabaseService entirely, since it's not adding any abstraction over Supabase. Just query Supabase directly here.